src/attribute_syncer.py

```diff
@@ -0,0 +1,556 @@
+                identity_store_id=ctx.identity_store_id,
+                s3_client=ctx.s3_client,
+                cache_config=ctx.cache_config,
+                managed_group_names=list(resolved_config.managed_group_names),
+            )
+            result.groups_processed = len(current_state)
+        except Exception as e:
+            logger.exception(f"Failed to fetch managed groups: {e}")
+            result.errors.append(f"Failed to fetch managed groups: {e}")
+            return _finalize_result(result)
+
+        # Step 5: Create sync state manager and compute actions
+        manager = SyncStateManager(
+            managed_group_ids=resolved_config.managed_group_ids,
+            mapper=mapper,
+            manual_assignment_policy=resolved_config.manual_assignment_policy,
+        )
+
+        actions = manager.compute_sync_actions(users, current_state)
+        logger.info(f"Computed {len(actions)} sync actions")
+
+        # Step 6: Execute actions
+        for action in actions:
+            try:
+                _execute_action(ctx, action, result)
+            except Exception as e:
+                logger.exception(f"Error executing action {action.action_type} for {action.user_email}: {e}")
+                result.errors.append(f"Error executing {action.action_type} for {action.user_email}: {e}")
+
+        # Step 7: Update cache with fresh data
+        try:
+            cache_module.set_cached_groups(ctx.s3_client, ctx.cache_config, all_groups)
+        except Exception as e:
+            logger.warning(f"Failed to update groups cache: {e}")
+
+        return _finalize_result(result)
+
+    except Exception as e:
+        logger.exception(f"Unexpected error during sync: {e}")
+        result.errors.append(f"Unexpected error: {e}")
+        return _finalize_result(result)
+
+
+def lambda_handler(event: dict[str, Any], context: object) -> dict[str, Any]:  # noqa: ARG001
+    """Lambda handler entry point for attribute sync.
+
+    **Feature: attribute-based-group-sync**
+    **Validates: Requirements 1.1, 1.2, 1.3, 1.4, 1.5, 5.3, 5.4, 5.5**
+
+    Args:
+        event: Lambda event (typically from EventBridge schedule).
+        context: Lambda context.
+
+    Returns:
+        Dictionary with sync operation result.
+    """
+    logger.info("Attribute syncer Lambda invoked", extra={"event": event})
+
+    # Load configuration
+    try:
+        config = load_sync_config()
+    except SyncConfigurationError as e:
+        logger.exception(f"Configuration error: {e}")
+        return {
+            "statusCode": 500,
+            "body": {"error": str(e), "success": False},
+        }
+
+    if not config.enabled:
+        logger.info("Attribute sync is disabled, skipping")
+        return {
+            "statusCode": 200,
+            "body": {"message": "Attribute sync is disabled", "success": True},
+        }
+
+    # Initialize clients
+    identity_store_client: IdentityStoreClient = boto3.client("identitystore")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

A Boto3 client is initialized from a Lambda handler. To speed up your Boto3 client initialization and minimize the operational cost of your Lambda function, we recommend that you create the client at the level of the module that contains the handler, and then reuse it between invocations.

[Learn more](https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html#function-code)

Similar issue at line number 484.